src/methods.ts

```diff
@@ -158,11 +159,18 @@ export async function upload(inputs: InputsArtifacts): Promise<void> {
   }
   const globber = await glob.create(inputs.source, globOptions)
   for await (const file of globber.globGenerator()) {
-    requests.push(upload_one_file(4, file, dirname, name, inputs))
+    requests.push(file)
   }
-  core.info('Waiting for all requests to finish')
-  await Promise.all(requests)
-  core.info('All requests are done')
+
+  core.info(requests.length.toString())
+
+  await async.eachLimit(requests, 16, (async (file: string, next) => {
```

Comment:
can the number of request `16` be like a variable set somewhere

Reply:
It could. But as it was only used there, I didn't bother doing it.